### DISCUSSION TITLE
Nextflow with Docker does not find local image

### DISCUSSION THREAD

**Discussion by alex-d13** [nextflow-io/nextflow#3410] [Q&A] [answered]
## Bug report 

I am using Nextflow with Docker and Slurm. I created a local Docker image called `omnideconv_benchmark` which I wanted to use to run my NF script. However, it seems like Nextflow does not find my image:

```
Command error:
  Unable to find image 'omnideconv_benchmark:latest' locally
  docker: Error response from daemon: pull access denied for omnideconv_benchmark, repository does not exist or may require 'docker login': denied: requested access to the resource is denied.
  See 'docker run --help'.
```

### Expected behavior and actual behavior

I would expect that the local image can be found and used by nextflow. Instead it is not found and the error above appears.

### Steps to reproduce the problem

-

### Program output 

[nextflow.log](https://github.com/nextflow-io/nextflow/files/10042910/nextflow.log)

### Environment 

* Nextflow version: [22.10.1] 
* Java version: [11.0.17]
* Operating system: [Ubuntu 20.04.5]
* Bash version: 5.0.17

### Additional context

I am starting Nextflow in a slurm spool directory. Maybe that could be the reason, that it does not find images out of its scope?

**Comment by pditommaso** [comment]
The error message says `repository does not exist or may require 'docker login'` 

This means you need to authenticate the docker client in the slurm nodes when the job running. Ask your sysadmin for details

**Reply by alex-d13** [reply]
Hi,

Yes I noticed that as well, but when I do use `docker login` prior to running nextflow successfully, i still get the same error. 
(Btw: i am working on a small cluster on my own, so there is no sysadmin apart from myself :D)

Could it be a solution to upload my image to docker hub and then nextflow will try to pull it from there?

**Comment by alex-d13** [comment, marked as answer]
I fixed this issue by uploading my image to Dockerhub and then nextflow will pull it from there.

Conversation can be closed :)
